src/Components/NavbarComponent/ProfileDrawerComponent/ProfileDrawerComponent.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import {
+  Drawer,
+  List,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+  type Theme,
+  useTheme,
+} from "@mui/material";
+import ManageAccountsIcon from "@mui/icons-material/ManageAccounts";
+import Brightness4Icon from "@mui/icons-material/Brightness4";
+import Brightness7Icon from "@mui/icons-material/Brightness7";
+import {COLORMODECONTEXT} from "../../../Scripts/ThemeContext";
+
+interface ProfileDrawerProps {
```

Comment:
This look like a common state/props, I think you should consider move it to a type file.